tests/functional/test_direct_edit.py

```diff
@@ -86,6 +87,53 @@ def test_cleanup_file(direct_edit):
     assert file.is_file()
 
 
+def test_corrupted_download(manager_factory, nuxeo_url):
+    manager, engine = manager_factory()
+
+    def corrupted_error_signal(label: str, values: List) -> None:
+        nonlocal received_corrupted
+        assert label == "DIRECT_EDIT_CORRUPTED_DOWNLOAD"
+        assert values == []
+        received_corrupted += 1
+
+    def failed_error_signal(label: str, values: List) -> None:
+        nonlocal received_failure
+        assert label == "DIRECT_EDIT_FAILURE"
+        assert values == []
+        received_failure = True
+
+    def corrupted_download(*_, **__):
+        raise CorruptedFile("test", "test", "test")
+
+    with manager:
+        received_corrupted = 0
+        received_failure = False
+
+        direct_edit = manager.direct_edit
+        direct_edit._folder.mkdir()
+
+        direct_edit.directEditError.connect(corrupted_error_signal)
+        direct_edit.directEditError.connect(failed_error_signal)
+
+        user = engine.remote_user
+        get_engine = direct_edit._get_engine
+
+        # Engine found, even with uppercase username
+        engine = get_engine(nuxeo_url, user=user)
+        assert engine
+
+        blob = Mock()
+        blob.digest = None
+
+        with patch.object(engine.remote, "download", new=corrupted_download):
```

Comment:
Try to use https://github.com/nuxeo/nuxeo-drive/blob/6605bde9cfb3c6dc519d89788742705eb2fb1445/tests/old_functional/__init__.py#L343 instead.